dev/PushNotifications/PushNotificationTelemetry.h

```diff
@@ -116,21 +120,61 @@ class PushNotificationTelemetry : public wil::TraceLoggingProvider
         return m_eventCount;
     }
 
-    wchar_t m_appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+    inline bool IsPackagedApp()
+    {
+        static const bool isPackagedApp = AppModel::Identity::IsPackagedProcess();
+
+        return isPackagedApp;
+    }
+
+    inline const wchar_t* GetAppName()
+    {
+        static const std::wstring appName = IsPackagedApp() ? GetAppNamePackaged() : GetAppNameUnpackaged();
+
+        return appName.c_str();
+    }
 
-    wchar_t* GetAppUserModelId()
+    std::wstring GetAppNamePackaged() noexcept
     {
-        if (m_appUserModelId[0] == '\0')
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+
+        UINT32 appUserModelIdSize = ARRAYSIZE(appUserModelId);
+        auto result = GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId);
+        if (result != ERROR_SUCCESS)
+        {
+            wcscpy_s(appUserModelId, L"AppUserModelId not found");
+            LOG_WIN32(result);
+        }
+
+        return appUserModelId;
+    }
+
+    PCWSTR CensorFilePath(_In_opt_ PCWSTR path) noexcept
```

Comment:
NIT: Remove `_In_opt_`

InOpt is redundant with PCWSTR

Reply:
done